### PR TITLE
test(Metadata-Controller): add tests for metadata endpoints

### PR DESCRIPTION
**Pull request by nicoprow**

### CODE REVIEW

File: src/test/kotlin/com/catenax/gpdm/controller/MetadataControllerIT.kt [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+        metadata.forEach { postMetadata(webTestClient, it) }
+
+        val returnedPage = getMetadataPage(webTestClient, 0, 2)
+
+        assertThat(returnedPage.totalElements).isEqualTo(expected.size.toLong())
+        assertThat(returnedPage.totalPages).isEqualTo(ceil(expected.size / 2f).toInt())
+        assertThat(returnedPage.page).isEqualTo(0)
+        assertThat(returnedPage.contentSize).isEqualTo(2)
+        assertThat(returnedPage.content.size).isEqualTo(2)
+        assertThat(returnedPage.content).containsAnyElementsOf(expected)
```
Comment (pazepaze) [inline src/test/kotlin/com/catenax/gpdm/controller/MetadataControllerIT.kt:287]: isSubsetOf seems more appropriate here

File: src/test/kotlin/com/catenax/gpdm/controller/MetadataControllerIT.kt [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        val expectedPage = PageResponse(expected.size.toLong(), 1, 0, expected.size, expected)
+
+        assertThat(returnedPage).usingRecursiveComparison().ignoringCollectionOrder().isEqualTo(expectedPage)
+    }
+
+    /**
+     * Given page with metadata
+     * When asking for second page
+     * Then return next metadata
+     */
```
Comment (pazepaze) [inline src/test/kotlin/com/catenax/gpdm/controller/MetadataControllerIT.kt:254]: Test does not just check the second page though. Adjust documentation accordingly.

**Review by pazepaze — CHANGES_REQUESTED** [review]

File: src/test/kotlin/com/catenax/gpdm/controller/MetadataControllerIT.kt [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+    @BeforeEach
+    fun beforeEach() {
+        testHelpers.truncateDbTables()
+    }
+
+    /**
+     * Given no identifier type
+     * When creating new identifier type
+     * Then created identifier type returned
+     */
```
[the remark this answers is not in the document]
Comment (nicoprow) [reply src/test/kotlin/com/catenax/gpdm/controller/MetadataControllerIT.kt]: done

File: src/test/kotlin/com/catenax/gpdm/controller/MetadataControllerIT.kt (hunk 1 above)
Comment (nicoprow) [reply src/test/kotlin/com/catenax/gpdm/controller/MetadataControllerIT.kt]: true, used that function instead

File: src/test/kotlin/com/catenax/gpdm/controller/MetadataControllerIT.kt (hunk 2 above)
Comment (nicoprow) [reply src/test/kotlin/com/catenax/gpdm/controller/MetadataControllerIT.kt]: made the documentation more matching

**Pull request merged**